src/Input/Input.js

```diff
@@ -69,11 +70,19 @@ export const styleSheet = createStyleSheet('MuiInput', (theme) => {
       },
     },
     disabled: {
-      color: theme.palette.text.disabled,
+      color: theme.palette.input.disabled,
       cursor: 'not-allowed',
     },
     underline: {
-      borderBottom: `1px solid ${theme.palette.text.divider}`,
+      borderBottom: `1px solid ${theme.palette.input.bottomLine}`,
+      '&:hover:not($disabled)': {
+        borderBottom: `2px solid ${theme.palette.text.primary}`,
+        marginBottom: -1,
+        transition: transitions.create('border-color', {
+          duration: transitions.duration.shorter,
+          easing: transitions.easing.ease,
+        }),
+      },
       '&$disabled': {
         borderBottomStyle: 'dotted',
```

Comment:
@oliviertassinari My concern it that using `background-image` we affect custom styling and using `border-image` things get complicated as there is a need to use `svg` image to maintain color and I could get it working with css3 gradient.